### PR TITLE
Make API more consistent with other smart pointer types

### PR DESCRIPTION
**Pull request by jacob-hughes**
This PR adds a few API improvements which allows `Rc` to be more easily swapped with `Gc` when used as a library.

### CODE REVIEW

**Comment by ltratt** [comment]
Please squash.

**Comment by jacob-hughes** [comment]
If there are no comments it can be merged as is :)

**Comment by ltratt** [comment]
The rustfmt commit should be squashed shouldn't it?

**Comment by jacob-hughes** [comment]
I made it a separate commit because it wasn't due to any of the changes in the previous commit. Rustfmt just wasn't tracking the `gc_test/` directory so it wouldn't break on the build, this is when I noticed it.

I've pushed a separate commit (74f0e6d0803fea4828a939347b8054438fa58cac) which adds rustfmt checking of  `gc_tests/` to bors. Let me know if you're happy with it and I'll squash them together.

**Comment by ltratt** [comment]
Please squash.

**Comment by ltratt** [comment]
Oops, hang on...

File: .buildbot.sh [hunk 1]
```diff
@@ -12,6 +12,6 @@ export PATH=`pwd`/.cargo/bin/:$PATH
 
 rustup component add --toolchain nightly rustfmt-preview || cargo +nightly install --force rustfmt-nightly
 
-cargo +nightly fmt --all -- --check
+cargo +nightly fmt --all -- --check gc_tests/tests/*
```
Comment (ltratt) [inline .buildbot.sh]: Does this mean we *only* check `gc_tests/tests/*`?

Comment (jacob-hughes) [reply .buildbot.sh]: No I've tested it, it's in addition to `-all`.

Comment (ltratt) [reply .buildbot.sh]: OK.

**Comment by ltratt** [comment]
Please squash.

**Comment by jacob-hughes** [comment]
Squashed

**Pull request merged**